From bee23159bb0c234203c27e42172542aa3b8c0ee1 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 06 Feb 2023 11:51:27 +0200
Subject: [PATCH] [SW-99614] gaudi2: security: remove dead code path

Functions hl_secure_lbw_rr(), hl_secure_hbw_rr(), and
hl_secure_hbw_common() have paths that are never executed in Ppboot and
cannot be garbage collected, so they are compiled with dead code inside.
Remove usage of those function in Ppboot by using hl_secure_lbw_rr_gen()
and hl_secure_hbw_pcie_rr() functions directly.
This removes the dead code paths for Ppboot.

Change-Id: Ib546c362c63aa9b8151a5df345b4a0f233755a2e
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index fcd8472..3453bf6 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -855,17 +855,14 @@
 static void ppboot_secure_lbw(void)
 {
 	u64 min, max;
-	struct hl_rr_eng_select rr_sel;
-
-	memset(&rr_sel, 0, sizeof(struct hl_rr_eng_select));
-
-	/* Block only pcie RR in ppboot */
-	rr_sel.pcie = 1;
 
 	/* CFG SPACE: 0x1000007FF8000000 - 0x1000007FFFFFFFFF */
 	min = CFG_BASE;
 	max = CFG_BASE + CFG_SIZE - 1;
-	hl_secure_lbw_rr(HL_SEC_LBW_DBG_CFG_SPACE, min, max, rr_sel);
+
+	/* Block only pcie RR in ppboot */
+	hl_secure_lbw_rr_gen(HL_SEC_LBW_DBG_CFG_SPACE, min, max,
+				0, HL_SEC_PCIE_INST_NUM, HL_SEC_ENGINE_PCIE);
 }
 
 static void preboot_secure_lbw(void)
@@ -1036,13 +1033,7 @@
 
 static void ppboot_secure_hbw(void)
 {
-	struct hl_rr_eng_select rr_sel;
 	u64 min, max;
-
-	memset(&rr_sel, 0, sizeof(struct hl_rr_eng_select));
-
-	/* Only block pcie RR in ppboot */
-	rr_sel.pcie = 1;
 
 	/*
 	 * SPI FLASH END to SP SRAM END
@@ -1051,9 +1042,23 @@
 	 */
 	min = SPI_FLASH_END;
 	max = SCRATCHPAD_SRAM_ADDR + SCRATCHPAD_SRAM_SIZE;
-	hl_secure_hbw_rr(HL_SEC_HBW_ICCM0_TO_SP_SRAM, min, max, rr_sel);
+	hl_secure_hbw_pcie_rr(HL_SEC_HBW_ICCM0_TO_SP_SRAM, min, max, true);
 
-	hl_secure_hbw_common(rr_sel);
+	/*
+	 * Block Reserved space to HBM
+	 * 0x1000008000000000 - 0x1000FFFFFFFFF000
+	 */
+	min = SRAM_RSVD_BASE;
+	max = SRAM_END_EXC_TOP4K;
+	hl_secure_hbw_pcie_rr(HL_SEC_HBW_SRAM, min, max, true);
+
+	/*
+	 * Block SPI Flash memory space
+	 * 0x1000007FF6000000 - 0x1000007FF7000000
+	 */
+	min = SPI_FLASH_BASE;
+	max = SPI_FLASH_END;
+	hl_secure_hbw_pcie_rr(HL_SEC_HBW_SPI_FLASH, min, max, true);
 }
 
 static void preboot_secure_hbw(void)
